onnxruntime/core/providers/cpu/controlflow/loop.cc

```diff
@@ -203,22 +203,31 @@ Status LoopImpl::Initialize() {
   }
 
   auto* max_trip_count_tensor = context_.Input<Tensor>(0);
-  auto iter_num_rank = max_trip_count_tensor ? max_trip_count_tensor->Shape().NumDimensions() : 0;
   auto* cond_tensor = context_.Input<Tensor>(1);
-  auto condition_rank = cond_tensor ? cond_tensor->Shape().NumDimensions() : 0;
 
-  if (condition_rank >= 2 || iter_num_rank >= 2) {
-    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
-                           "'Loop' input 'M' and 'cond' should be a scalar tensor, but have ranks of ",
-                           condition_rank, " and ", iter_num_rank);
+  if (max_trip_count_tensor) {
+    if (max_trip_count_tensor->Shape().Size() != 1) {
```

Comment:
Since we are using subgraph's rank, should we test subgraph here instead as well?